tests/test_model.py

```diff
@@ -1501,6 +1498,141 @@ def test_add_boundary_condition_by_geometry_ids(self, create_default_3d_soil_mat
+
+        # create a 3D model
+        model = Model(ndim)
+        model.extrusion_length = 1
+
+        # set expected parameters of the load conditions
+        expected_0d_model_part_parameters = PointLoad(active=[False, True, False], value=[0, -200, 0])
+        expected_1d_model_part_parameters = LineLoad(active=[False, True, False], value=[0, -20, 0])
+        expected_2d_model_part_parameters = SurfaceLoad(active=[False, True, False], value=[0, -2, 0])
+        moving_load_parameters = MovingLoad(
+            origin=[0, 1, 0.5],
+            load=[0.0, -10.0, 0.0],
+            velocity=5.0,
+            offset=3.0,
+            direction=[1, 1, 1]
+        )
+
+        # add body model part
+        soil_material = create_default_3d_soil_material
+        model.add_soil_layer_by_coordinates([(0, 0, 0), (1, 0, 0), (1, 1, 0), (0, 1, 0)], soil_material, "test_soil")
+
+        # add boundary conditions in 0d, 1d and 2d
+        model.add_load_by_geometry_ids([3, 4, 7, 8], create_default_point_load_parameters, "point_loads")
+        model.add_load_by_geometry_ids([3, 8, 10, 11], create_default_line_load_parameters, "line_loads")
+        model.add_load_by_geometry_ids([4], create_default_surface_load_parameters, "surface_load")
+        model.add_load_by_geometry_ids([3, 8, 10], moving_load_parameters, "moving_load")
+
+        # set expected geometry 0d load
+        expected_load_points = {3: Point.create([1, 1, 0], 3), 4:  Point.create([0, 1, 0], 4),
+                                    7: Point.create([1, 1, 1], 7), 8:  Point.create([0, 1, 1], 8)}
+        expected_load_geometry_0d = Geometry(expected_load_points, {}, {}, {})
+
+        # set expected geometry 1d load
+        expected_load_lines = {3: Line.create([3, 4], 3), 8: Line.create([3, 7], 8),
+                                   10: Line.create([4, 8], 10), 11: Line.create([7, 8], 11)}
+
+        expected_load_geometry_1d = Geometry(expected_load_points, expected_load_lines, {}, {})
+
+        # set expected geometry 2d load
+        expected_load_geometry_2d = Geometry()
+        expected_load_geometry_2d.points = {
+            3: Point.create([1, 1, 0], 3), 7: Point.create([1, 1, 1], 7),
+            8:  Point.create([0, 1, 1], 8), 4:  Point.create([0, 1, 0], 4)
+        }
+        expected_load_geometry_2d.lines = {
+            8: Line.create([3, 7], 8), 11: Line.create([7, 8], 11),
+            10: Line.create([4, 8], 10), 3: Line.create([3, 4], 3)
+        }
+        expected_load_geometry_2d.surfaces = {
+            4: Surface.create([8, 11, -10, -3], 4)
+        }
+        # collect all expected geometries
+        all_expected_geometries = [expected_load_geometry_0d, expected_load_geometry_1d, expected_load_geometry_2d]
+
+        for expected_geometry, model_part in zip(all_expected_geometries, model.process_model_parts):
+
+            TestUtils.assert_almost_equal_geometries(expected_geometry, model_part.geometry)
+
+        # check 0d parameters
```

Comment:
I would change these comments and parameter names refering to the load type rather than the geometry dimension. In this case " Check point load parameters " and  npt.assert_allclose(model.process_model_parts[0].parameters.value, expected_point_load_parameters.value)